### ISSUE TITLE
Replace hbase-testing-util with docker container

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#2295]
`hbase-testing-util` seems to be a rather large bucket of stuff, much of which causes issues. Notably the presence of Jersey JAX-RS classes, which Quarkus RestEasy attempts to operate on. This is failing on the `quarkus-master` branch.

```
Caused by: java.lang.RuntimeException: Unable to handle class: com.sun.jersey.api.core.ResourceConfig
	at io.quarkus.resteasy.server.common.deployment.ResteasyServerCommonProcessor.getAllowedClasses(ResteasyServerCommonProcessor.java:979)
	at io.quarkus.resteasy.server.common.deployment.ResteasyServerCommonProcessor.build(ResteasyServerCommonProcessor.java:216)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Maybe it'd be better to replace `hbase-testing-util` with a container based test?

**Comment by JiriOndrusek** [comment]
@jamesnetherton  I investigated the possibility of using testcontainer with hbase test. It could work successfully but only with a few restrictions:
1 - Ports can not be random (2181, 16000, 16020) - This is caused by the fact, that client asks zookeeper about the hbase master location. Zookeeper knows only about ports used internally in docker network. If I use random ports (mapped via docker), zookeeper still returns the original internal ports and client can not connect to them.
2 - For the same reason as above, there has to be a modification of `hosts` to contain a rule for hbase hostname. (without it, client gets from zookeeper location with unknown host)

To summarize it:

- If some of the ports 2181, 16000, 16020 is already taken, test will fail. 
- If there is no rule `127.0.0.1 hbase` in /etc/hosts, test will fail. 

I'm trying to at least rid of the second one restriction, but I think that it won't be possible.

Do you think, that it could be possible to require a rule in `hosts` for the test?

**Comment by jamesnetherton** [comment]
Thanks @JiriOndrusek for looking into this!

It'd be nice to avoid fixed ports, but if it's not possible, then it's ok. Some of the other tests require fixed ports.

I think we definitely need to avoid relying on host entry hacks, as everyone wanting to run the test suite will need it. Have you tried configuring the container to use host networking? You may not need such hacks in that mode.

**Comment by JiriOndrusek** [comment]
Thanks for the advice, network `host` is exactly what I was looking for. Host entry no longer needed.
I'll try to find how to avoid fixed ports in case it would be possible.
